TODO

```diff
@@ -29,8 +26,34 @@ Janitorial Clean-ups:
 
 Features:
 
-* replace all canonicalize_file_name() invocations by chase_symlinks(), in
-  particulr those where a rootdir is relevant.
+* set ProtectSystem=strict for all our usual services.
+
+* maybe add gpt-partition-based user management: each user gets his own
+  LUKS-encrypted GPT partition with a new GPT type. A small nss module
+  enumerates users via udev partition enumeration. UIDs are assigned in a fixed
+  way: the partition index is added as offset to some fixed base uid. User name
+  is stored in GPT partition name. A PAM module authenticates the user via the
+  LUKS partition password. Benefits: strong per-user security, compatibility
+  with stateless/read-only/verity-enabled root.
+
+* gpt-auto logic: introduce support for discovering /var matching an image. For
+  that, use a partition type UUID that is hashed from the OS name (as encoded
+  in /etc/os-release), the architecture, and 4 new bits from the gpt flags
+  field of the root partition. This way can easily support multiple OS
+  installations on the same GPT partition table, without problems with
+  unmatched /var partitions.
+
+* gpt-auto logic: related to the above, maybe support a "secondary" root
+  partition, that is mounted to / and is writable, and where the actual root's
+  /usr is mounted into.
+
+* add dm-verity boots, and in nspawn (libcryptsetup knows this, should be relatively straight-forward)
+
+* machined: add apis to query /etc/machine-info data of a container
+
+* .mount and .swap units: add Format=yes|no option that formats the partition before mounting/enabling it, implicitly
```

Comment:
Unfortunately mkfs does not work like this. It only understands `-t`, and all other options are passed to the individual fsck implementations, and there's no standardization there. It would be great if mkfs provided uniform options, but it doesn't, even for -U and -L. It might be necessary to work with util-linux to provide this first.

I *do* think this could be really useful, I just worry that although it is fairly easy to provide a basic implementation, it's much harder to graduate to a full-blown implementation that users will want because of missing standardization of tools. tmpfs is a special case, because you don't care too much about the fs type, and even less about options. If this is only designed to reimplement that, just at part of mount units and not cryptsetup, and only with one or two fs types, then OK. But if this is supposed to be more general, then it's harder to make that happen without encoding a lot of fs-specific knowledge.